armi/physics/neutronics/crossSectionGroupManager.py

```diff
@@ -1235,10 +1235,13 @@ def createRepresentativeBlocksUsingExistingBlocks(
 
             # create a new block collection that inherits all of the properties
             # and settings from oldBlockCollection.
-            if len(oldBlockCollection._validRepresentativeBlockTypes) > 0:
-                validBlockTypes = []
-                for flag in oldBlockCollection._validRepresentativeBlockTypes:
-                    validBlockTypes.append(flags._toString(Flags, flag))
+            if oldBlockCollection._validRepresentativeBlockTypes is not None:
+                if len(oldBlockCollection._validRepresentativeBlockTypes) > 0:
+                    validBlockTypes = []
+                    for flag in oldBlockCollection._validRepresentativeBlockTypes:
+                        validBlockTypes.append(flags._toString(Flags, flag))
+                else:
+                    validBlockTypes = None
             else:
                 validBlockTypes = None
```

Comment:
```suggestion
            validBlockTypes = oldBlockCollection._validRepresentativeBlockTypes
            if validBlockTypes is not None and len(validBlockTypes) > 0:
                validBlockTypes = [
                    flags._toString(Flags, flag)
                    for flag in oldBlockCollection._validRepresentativeBlockTypes
                ]
```

Reply:
I won't die on this hill, I just thought reducing the logic here to one `if`-statement made it easier to read.

Up to you!

Reply:
This is a good simplification. `validBlockTypes` might end up being `[]` instead of `None`, but the behavior will be the same when it is passed into the `BlockCollection` constructor.